spinta/utils/response.py

```
@@ -106,7 +108,73 @@
+                    detail="missing authorization header",
+                )
+
+            if not auth_header.startswith('Bearer '):
+                raise HTTPException(
+                    status_code=403,
+                    detail="invalid authorization header",
+                )
+
+            jwt_token = auth_header.strip('Bearer ') if auth_header else None
+
+            # TODO: check for special scope which allows id creation
+            # scopes = jwt.decode(jwt_token)
+            #
+            # XXX: currently for dev purposes and testing use
+            # fake token `f00b4rb4z`
+            can_create_ids = bool(jwt_token == "f00b4rb4z")
+
+            # make sure json is valid
+            try:
+                data = await request.json()
+            except json.decoder.JSONDecodeError:
+                raise HTTPException(
+                    status_code=400,
+                    detail="not a valid json",
+                )
+
+            object_is_being_created = ('id' in data.keys() and not params.id)
+            if object_is_being_created:
+                if not can_create_ids:
+                    raise HTTPException(
+                        status_code=400,
+                        detail="cannot create 'id'",
+                    )
+                else:
+                    try:
+                        commands.get(context, model, model.backend, data['id'])
+                    except (MultipleRowsException, NoResultsException):
+                        pass
+                    else:
+                        # XXX: should raise HTTPException here,
+                        # but for some reason in tests, when raising
+                        # HTTPException here, the database connection
+                        # closes sooner than the test can finish,
+                        # so I return just a JSONResponse with HTTP/400
+                        return JSONResponse({"error": "cannot create duplicate 'id'"},
```

`raise` instead of `return`?